src/test/scala/uk/gov/hmrc/test/ui/pages/returns/ContactNamePage.scala

```diff
@@ -32,6 +32,34 @@ object ContactNamePage extends BasePage {
     this
   }
 
+  def provideContactName(contactName: String): this.type = {
+    SharedActions
+      .enterDetails(contactName)
+    submitPage()
+    this
+  }
+
+  def provideContactRole(contactRole: String): this.type = {
+    SharedActions
+      .enterDetails(contactRole)
+    submitPage()
+    this
+  }
+
+  def provideContactEmailAddress(emailAddress: String): this.type = {
+    SharedActions
+      .enterDetails(emailAddress)
+    submitPage()
+    this
+  }
+
+  def provideContactNumber(contactNumber: String): this.type = {
+    SharedActions
+      .enterDetails(contactNumber)
+    submitPage()
+    this
+  }
```

Comment:
shouldn't all these methods contain `onPage(....heading)` to assert they go to the right place?